core/modules/menu/menu.admin.inc

```diff
@@ -412,74 +412,46 @@ function menu_edit_item($form, &$form_state, $type, $item, $menu) {
     '#description' => t('When displayed in a "hierarchical tree" menu style, children of expanded menu links will always be visible.'),
   );
 
-  $separate_menu = config_get('menu.settings', 'menu_separate_menu_selector');
   $menu_name = $item['menu_name'];
   $menus = menu_get_menus();
-  if ($separate_menu) {
-    $form['parent_menu'] = array(
-      '#type' => 'select',
-      '#title' => t('Menu'),
-      '#default_value' => $menu_name,
-      '#options' => $menus,
-      '#description' => t('Select the menu that this menu link should belong to.'),
-      '#attributes' => array('class' => array('parent-menu-select')),
-      '#ajax' => array(
-        'callback' => 'menu_parent_menu_callback',
-      ),
-    );
-    if (isset($form_state['values']['parent_menu'])) {
-      $menu_name = $form_state['values']['parent_menu'];
-    }
-    // Generate a list of possible parents (not including this link or
-    // descendants) from the selected menu.
-    $options = menu_parent_options(array($menu_name => $menus[$menu_name]), $item);
-    $options[array_keys($options)[0]] .= ' ' . t('(top level)');
-  }
-  else {
-    // Generate a list of possible parents (not including this link or
-    // descendants) from all menus.
-    $options = menu_parent_options($menus, $item);
-  }
+  $form['parent_menu'] = array(
+    '#type' => 'select',
+    '#title' => t('Menu'),
+    '#default_value' => $menu_name,
+    '#options' => $menus,
+    '#description' => t('Select the menu that this menu link should belong to.'),
+    '#attributes' => array('class' => array('menu-menu-select')),
+    '#ajax' => array('callback' => 'menu_parent_menu_callback'),
+  );
+  if (isset($form_state['values']['parent_menu'])) {
+    $menu_name = $form_state['values']['parent_menu'];
+  }
+  // Generate a list of possible parents (not including this link or
+  // descendants) from the selected menu.
+  $options = menu_parent_options(array($menu_name => $menus[$menu_name]), $item);
+  $option_first_key = array_keys($options)[0];
```

Comment:
This works, but there are "nicer" ways to do it with some in-built PHP functions: https://stackoverflow.com/questions/1921421/get-the-first-element-of-an-array (and don't use `array_key_first()` since that is only supported in PHP 7.3+)

Reply:
Definitely nicer:

```
  $options = menu_parent_options(array($menu_name => $menus[$menu_name]), $item);
  $option_first_key = key($options);
```